### DISCUSSION TITLE
Create action using anything but JavaScript/TypeScript?

### DISCUSSION THREAD

**Discussion by ALIENQuake** [community/community#25530] [Actions]
<p>Hello,</p>
<p>It’s possible to create an action using anything but JavaScript/TypeScript? Like for eg: PowerShell/Python or c#/Java?</p>

**Comment by peter-evans** [comment, marked as answer]
<p>The only options right now are Docker container actions or Javascript/Typescript. What you can do though, is use the <code>@actions/exec</code> package of <code>actions/toolkit</code> and execute tools and scripts in other languages.</p>
<p><a href="https://github.com/actions/toolkit/tree/master/packages/exec" rel="nofollow noopener">https://github.com/actions/toolkit/tree/master/packages/exec</a></p>
<p>I’ve made a template for a Python action using this method.</p>
<p><a href="https://github.com/peter-evans/python-action" rel="nofollow noopener">https://github.com/peter-evans/python-action</a></p>

**Comment by jherico** [comment]
<p>Unfortunately, such docker based actions <em>only</em> work on Linux, so if you’re doing builds on Mac or Windows for any reason, then they’re useless.  </p>
<p>I too would really like an alternative language for actions that also works on Windows or Mac.</p>

**Comment by peter-evans** [comment]
<p>Exactly. That was one of the main reasons why I made the Python action template based on the method I described. It’s currently the only way to make it multi-platform.</p>
<p>I reached out to GitHub support about the situation. They couldn’t tell me if there are other languages other than Javascript in the works, but what they did say is “We have no plans to support Windows or Mac Docker containers.”</p>
<p>I think they will bring out other action runners at some point. Perhaps when Actions and the Javascript <code>actions/toolkit</code> has matured a little more.</p>

**Comment by jherico** [comment]
<p>Ah, I see, I misunderstood the thrust of your comment.  I will take a look at this template, it sounds promising.</p>
